src/NServiceBus.Core.Tests/Recoverability/DefaultRecoverabilityPolicyTests.cs

```diff
@@ -3,41 +3,54 @@
     using System;
     using System.Collections.Generic;
     using System.IO;
-    using Transport;
     using NUnit.Framework;
+    using Transport;
 
     [TestFixture]
     public class DefaultRecoverabilityPolicyTests
     {
         [Test]
-        public void When_max_immediate_reties_have_not_been_reached_should_return_immediate_retry()
+        public void When_failure_is_caused_by_deserialization_exception_should_move_to_error()
+        {
+            var policy = CreatePolicy(maxImmediateRetries: 3, maxDelayedRetries: 3);
+            var errorContext = CreateErrorContext(numberOfDeliveryAttempts: 1, exception: new MessageDeserializationException("failed"));
+
+            var recoverabilityAction = policy(errorContext);
+
+            Assert.IsInstanceOf<MoveToError>(recoverabilityAction, "Should move deserialization exception directly to error.");
+        }
+
+        [Test]
+        public void When_max_immediate_retries_have_not_been_reached_should_return_immediate_retry()
         {
             var policy = CreatePolicy(maxImmediateRetries: 3);
             var errorContext = CreateErrorContext(numberOfDeliveryAttempts: 2);
 
-            var recoverabilityAction = policy.Invoke(errorContext);
+            var recoverabilityAction = policy(errorContext);
 
             Assert.IsInstanceOf<ImmediateRetry>(recoverabilityAction, "We should have one immediate retry left. It is second delivery attempt and we configured immediate reties to 2.");
         }
 
         [Test]
-        public void When_max_immediate_retries_exceeded_should_return_delayed_retry()
+        [TestCase(0)]
+        [TestCase(1)]
```

Comment:
what's the reason for the testcases here? they don't even seem to test edge cases like 2/3?